### ISSUE TITLE
Restart environment after importing from clipboard

### ISSUE THREAD

**Issue by feraguiba** [mockoon/mockoon#209]
When you have already started an environment and then you import a route from clipboard, I always forget to stop and then start again the environment. But... If I have imported something in an already started environment... It should be already available, right?

**Comment by KryptKode** [comment]
Yes. I noticed this too. It no longer  persists the environments when you close the application. It did so in previous versions

**Comment by 255kb** [comment]
@feraguiba adding, importing (file or clipboard) or duplicating a route, requires an environment restart
Would you like Mockoon to automatically restart? That could be a new option.

**Comment by feraguiba** [comment]
> @feraguiba adding, importing (file or clipboard) or duplicating a route, requires an environment restart
> Would you like Mockoon to automatically restart? That could be a new option.

Yes, it would be nice. Or at least the "stop" environment button should be changed to "restart needed" when you paste from clipboard. It doesn't change nowadays.

**Comment by 255kb** [comment]
I just tested on v1.6.0 and indeed icon changes to "restart needed". 
I corrected a bug where this wasn't the case especially with duplication. Maybe it solved it for clipboard import too.

**Comment by KryptKode** [comment]
@255kb is it normal for environments to not be persisted after the program is closed? It used to be persisted but on recent updates, it's not so I have to export  the environments to a file and re-import every time I open the program

**Comment by feraguiba** [comment]
> I just tested on v1.6.0 and indeed icon changes to "restart needed".
> I corrected a bug where this wasn't the case especially with duplication. Maybe it solved it for clipboard import too.

It is solved also for clipboard import! Then I launch the next request... Autorestart when importing from clipboard. Althought not so important as now you see clearly the environment must be restarted.

Thanks for your support @255kb

**Comment by feraguiba** [comment]
> @255kb is it normal for environments to not be persisted after the program is closed? It used to be persisted but on recent updates, it's not so I have to export the environments to a file and re-import every time I open the program

I have just tested it and the environment is persisted (Mac version 1.6)

**Comment by 255kb** [comment]
@KryptKode not at all. It should persist as before. Please open a new issue with maximum information so we can track this.

**Comment by 255kb** [comment]
@feraguiba ok, I will rename this issue to add an auto restart feature in the future

**Comment by gabrielperes97** [comment]
Hi again @255kb . Can we add the auto restart feature to any restart require case?  Should we add a config parameter to enable the auto restart?

**Comment by gabrielperes97** [comment]
I can work on this feature. Can you assign it to me?

**Comment by 255kb** [comment]
Hi @gabrielperes97 Sure you can work on it, I will assign it to you. 
Regarding the implementation here is what I have in mind: 
- like you said, a new setting to enable the auto-restart. Off by default. (do not forget to add a small settings "migration" in the settings.service)
- once the reducer mark the API with `needRestart` we should react to it, so probably by subscribing to this part of the store in the environments.service
- add tests in `.\test\suites\environment-restart-triggers.spec.ts`, mainly a new case where we open the settings, enable the auto-restart, start the env, perform an action that needs a restart, check that the restart is not needed (no orange rotation arrow). 

As always, keep it simple :)

**Comment by 255kb** [comment]
This feature will be abandoned in favor of #2078, which is a better solution overall.
